### DISCUSSION TITLE
Site Rules updates & Static pages / Обновления текста правил и статические страницы сайта

### DISCUSSION THREAD

**Discussion by paul-k-pastvu** [PastVu/pastvu#249] [General / Основное]
Сейчас текст правил на сайте реализован как статичная страница, отображаемая пользователю во всплывающем окне.
См. https://pastvu.com/rules

Был создан отдельный репозитарий в GitHub Pages под "статический" контент (помощь, правила, т.д.) и подключить GitHub Pages (можно даже домен pastvu.com привязать). Можно было бы вносить изменения через PR, отслеживать историю, оперативно менять что либо в тексте. 
См. https://github.com/PastVu/pages и отображение https://pages.pastvu.com

Здесь обсуждается внедрение этого репозитория в интерфейс основного сайта.

Связанные темы в новостях (4.11.2015) 
- https://pastvu.com/news/130
- https://pastvu.com/news/141?hl=comment-2571156

Связанные задачи: https://github.com/PastVu/pastvu/issues/288, https://github.com/PastVu/pastvu/issues/180

**Comment by paul-k-pastvu** [comment]
Не правила, но рядом: https://github.com/PastVu/pastvu/issues/252

**Reply by kabalin** [reply]
#288 тоже

**Reply by paul-k-pastvu** [reply]
И #180

**Comment by aeifn** [comment]
На самом деле, правила меняются не настолько часто и не таким большим количеством людей, чтобы делать под них отдельный пункт в админке.
Достаточно напрямую в браузере править файл https://github.com/PastVu/pastvu/blob/master/views/module/diff/rules.pug, или же при помощи pull-request.
К тому же, этот способ уже доставлят версионирование текста.

**Reply by paul-k-pastvu** [reply]
@aeifn @klimashkin - я правильно понимаю, что https://github.com/PastVu/pastvu/blob/master/views/module/diff/rules.pug - это текущая версия, причем этот текст сейчас отображается и в русской, и в английской версии сайта?

**Reply by kabalin** [reply]
Да. Для английской отдельный бранч, но текст там тоже русский: https://github.com/PastVu/pastvu/blob/en/views/module/diff/rules.pug Английская версия что на [pages](https://pages.pastvu.com/en/rules), это через автопереводчик пропущено, но вроде бы хорошо перевел, хотя не мешало бы внимательно перечитать.

**Comment by kabalin** [comment]
Предлагаю правила в GitHub Pages перенести. Создать отдельный репозитарий под "статический" контент (помощь, правила, т.д.) и подключить GitHub Pages (можно даже домен pastvu.com привязать). Можно было бы вносить изменения через PR, отслеживать историю, оперативно менять что либо в тексте.

**Reply by aeifn** [reply]
сделал репозиторий https://github.com/PastVu/rules

**Reply by kabalin** [reply]
Может `pages` назвать, чтобы туда не только правила, но и другой контент в перспективе.

**Reply by aeifn** [reply]
Переименовал

**Reply by paul-k-pastvu** [reply]
Можно сделать пример страницы и её отображение на тестовом сайте?

**Reply by aeifn** [reply]
> Можно сделать пример страницы и её отображение на тестовом сайте?

примерно так: https://pastvu.github.io/pages/rules

**Reply by kabalin** [reply]
> примерно так: https://pastvu.github.io/pages/rules

Выглядит торжественно )) 

На самом деле там выбор тем большой, к примеру можно взять популярную `minima` (пример https://jekyll.github.io/minima/) и [подкрутить ее](https://github.com/jekyll/minima/#customizing-templates) по цветовой гамме (чтобы советовала сайту pastvu).

**Reply by kabalin** [reply]
Вот инструкция как добавить мультиязычность может пригодится: https://jerry901.github.io/en/2020/02/2020-02-06-01/

**Reply by kabalin** [reply]
Починил правила (чтобы нумерация соответствовала сайту) и добавил тему: <del>https://kabalin.github.io/pastvu-pages/ru/rules</del>

Изменения в PR: https://github.com/PastVu/pages/pull/1

**UPD**: спасибо @aeifn, все изменения уже тут https://pastvu.github.io/pages/ru/rules

**Reply by paul-k-pastvu** [reply]
Два вопроса:
1) как это редактировать через веб?
2) как это будет в итоге прикручено к сайту? Не хотелось бы, чтобы это было на другом домене.

Собственно, у нас же есть тестовая версия. Давайте на ней попробуем вывести ссылки на статические страницы. Ссылка на правила в интерфейсе есть (но она, кстати, должна быть не в подвале, а в шапке сайта). Давайте её переделаем, чтобы она вела на новую статическую страницу.
И там же нужна новая ссылка "Помощь" (Help), которая будут вести на оглавление справочного раздела.

Если всё получится, дальше будум развивать этот блок и обсудим нужность выделения таких страниц по регионам и темам.

**Reply by paul-k-pastvu** [reply]
И хотелось бы услышать мнение от @klimashkin !

**Reply by aeifn** [reply]
> 1. как это редактировать через веб?

форкнуть репозиторий на гитхабе, внести изменения в веб-редакторе на гитхабе, сделать пулл-реквест.
или, если есть прямой доступ к репозиторию: просто внести изменения в веб-редакторе.


> 2. как это будет в итоге прикручено к сайту? Не хотелось бы, чтобы это было на другом домене.

например, на домене pages.pastvu.com

**Reply by kabalin** [reply]
>форкнуть репозиторий на гитхабе, внести изменения в веб-редакторе на гитхабе, сделать пулл-реквест.
или, если есть прямой доступ к репозиторию: просто внести изменения в веб-редакторе.

Да. В коммите желательно указать причину изменения и сослаться на тикет (если есть).

>например, на домене pages.pastvu.com

или docs.pastvu.com

Ссылку "Помощь" предлагаю добавить на сайт когда будет хотя бы одна статическая страница помощи 😃 

Еще надо мультиязычность добавить, попробую на выходных, нашел пару примеров как это сделать.

**Reply by kabalin** [reply]
Добавлена мультиязычность: https://pastvu.github.io/pages/ Инструкция по созданию страниц находится в README. Кстати, не знал что [английской версии правил не существует](https://github.com/PastVu/pastvu/blob/en/views/module/diff/rules.pug), пришлось для демонстрации концепции многоязычности скопировать пока русскую.

**Comment by paul-k-pastvu** [comment]
По поводу внутренней организации правил - хороший пример из сходного по идеологии проекта: http://www.plantarium.ru/page/help.html

**Reply by mkgrgis** [reply]
Отличный пример я бы сказал! Понятность и быстрота выхода на какой-то вопрос образцовые.

**Comment by paul-k-pastvu** [comment]
Вот нам ещё ссылок накидали  в справочный раздел - https://pastvu.com/news/141?hl=comment-2571156
А по архивам новостей этого - россыпи.

**Reply by kabalin** [reply]
Да, наверное вики-платформа была бы более удобным вариантом для обычных пользователей.

**Reply by paul-k-pastvu** [reply]
К тому же у нас в сообществе есть сразу несколько вики-админов готовых...
Но это громоздкая отдельная штука, не обязательно её сразу запускать. Вполне можно сначала переключиться на статические страницы. 
Нужно, чтобы:
1) в интерфейсе осноного сайта появилась ссылка на справочный раздел (причем отдельно - на русскую и ангдийскую версию)
2) ссылка на текст правил во всплывающем блоке (в подвале сайта)
  - либо меняется на статическую ссылку на https://pages.pastvu.com/ru/rules / https://pages.pastvu.com/en/rules
  - либо динамически отображает текущую версию этой страницы (чтобы они не "разъехались" вдруг)

После этого я публикую новость с анонсом и начинаю обсуждение структуры и наполнения нового раздела.

**Reply by kabalin** [reply]
> Но это громоздкая отдельная штука, не обязательно её сразу запускать. 

Да вики поднять то не сложно на swarm, были бы ресурсы и время настроить сервис и бэкап. Надо все же решить, чтобы не переделывать, на чем мы остановимся. Какой смысл делать структуру в markdown, если рано или поздно придется переносить все на вики (дополнительная работа).

**Comment by kabalin** [comment]
Документация вынесена в отдельный проект (https://docs.pastvu.com/), таким образом необходимость в редактировании статических страниц внутри приложения отпадает. Для добавления отдельных страниц помощи и прочей информации предлагаю описывать задачи в https://github.com/PastVu/docs
